docs/ReferenceArrayInput.md

```diff
@@ -7,103 +7,332 @@ title: "The ReferenceArrayInput Component"
+
+```jsx
+const filters = [
+    <ReferenceArrayInput label="Post tags" source="tags_ids" reference="tags">
+        <AutocompleteArrayInput label="Post tags" />
+    </ReferenceArrayInput>,
+];
+```
+
+## `parse`
+
+By default, children of `<ReferenceArrayInput>` transform the empty form value (an empty string) into `null` before passing it to the `dataProvider`. 
+
+If you want to change this behavior, you have to pass a custom `parse` prop to the `<ReferenceArrayInput>` *child component*, because  **`<ReferenceArrayInput>` doesn't have a `parse` prop**. It is the responsibility of the child component to parse the input value.
+
+For instance, if you want to transform an option value before submission, and the selection control is an `<AutocompleteArrayInput>` (the default), set [the `<AutocompleteArrayInput parse>` prop](./Inputs.md#parse) as follows:
+
+```jsx
+import { ReferenceArrayInput, AutocompleteArrayInput } from 'react-admin';
+
+<ReferenceArrayInput source="tags_ids" reference="tags">
+    <AutocompleteArrayInput parse={value => value === 'not defined' ? null : value} />
+</ReferenceArrayInput>
+```
+
+The same goes if the child is a `<SelectArrayInput>`:
+
+```jsx
+import { ReferenceArrayInput, SelectArrayInput } from 'react-admin';
+
+<ReferenceArrayInput source="tags_ids" reference="tags">
+    <SelectArrayInput parse={value => value === 'not defined' ? undefined : null} />
+</ReferenceArrayInput>
+```
+
+## `perPage`
+
+By default, `<ReferenceArrayInput>` fetches only the first 25 values. You can extend this limit by setting the `perPage` prop.
+
+```jsx
+<ReferenceArrayInput source="tags_ids" reference="tags" perPage={100} />
+```
+
+This prop is mostly useful when using [`<SelectArrayInput>`](./SelectArrayInput.md) or [`<CheckboxGroupInput>`](./CheckboxGroupInput.md) as child, as the default `<AutocompleteArrayInput>` child allows to filter the possible choices with a search input.
+
+## `reference`
+
+The name of the reference resource. For instance, in a post form, if you want to edit the post tags, the reference should be "tags".
+
+```jsx
+<ReferenceArrayInput source="tags_ids" reference="tags" />
+```
+
+`<ReferenceArrayInput>` will use the reference resource [`recordRepresentation`](./Resource.md#recordrepresentation) to display the selected record and the list of possible records. So for instance, if the `tags` resource is defined as follows:
+
+```jsx
+<Resource name="tags" recordRepresentation="name" />
+```
+
+Then `<ReferenceArrayInput>` will display the company name in the input and the list of possible values.
+
+You can override this default by specifying the `optionText` prop in the child component. For instance, for an `<AutocompleteArrayInput>`:
+
+```jsx
+<ReferenceArrayInput source="tags_ids" reference="tags">
+    <AutocompleteArrayInput optionText="reference" />
+</ReferenceArrayInput>
+```
+
+## `queryOptions`
```

Comment:
Actually, this is not true, queryOptions prop is part of the ReferenceInput (https://github.com/marmelab/react-admin/pull/8192) but not yet of the ReferenceArrayInput